services/npm/npm.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const BaseService = require('../base');
+const {
+  checkErrorResponse,
+  asJson,
+} = require('../../lib/error-helper');
+const {
+  makePackageDataUrl,
+  typeDefinitions,
+} = require('../../lib/npm-badge-helpers');
+
+module.exports = class NPMTypeDefinitions extends BaseService {
+  static get category() {
+    return 'version';
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'type definitions' };
+  }
+
+  static get url() {
+    return {
+      base: 'npm/types',
+      format: '(?:@([^/]+)/)?([^/]+)',
+      capture: ['scope', 'packageName'],
+      queryParams: ['registry_uri'],
+    };
+  }
+
+  static get examples() {
+    return [{
+      title: 'npm type definitions',
+      previewUrl: 'chalk',
+      keywords: ['node', 'typescript', 'flow'],
+    }];
+  }
+
+  async handle({ scope, packageName }, { registry_uri: registryUrl }) {
+    const apiUrl = makePackageDataUrl({ registryUrl, scope, packageName });
+
+    const json = await this._sendAndCacheRequest(apiUrl, {
+      headers: { 'Accept': '*/*' },
+    }).then(checkErrorResponse.asPromise({ notFoundMessage: 'package not found' }))
+      .then(asJson);
```

Comment:
Now that I've merged #1702 you should be able to rebase this on to master and modify this to extend `BaseJsonService` and replace this block with `_requestJson()`. Apart from that, looks good :+1: